lib/miq_expression.rb

```diff
@@ -344,6 +344,15 @@ def sql_supports_atom?(exp)
         # TODO: Support includes operator for sub-sub-tables
         return false
       end
+    when "includes any", "includes all", "includes only"
+      # Support this only from the main model (for now)
+      if exp[operator].keys.include?("field") && exp[operator]["field"].split(".").length == 1
+        model, field = exp[operator]["field"].split("-")
+        method = "miq_expression_#{operator.downcase.tr(' ', '_')}_#{field}_arel"
```

Comment:
Aside: I feel like we'll never know we should implement this method for other models if they're having the same problem.  We probably don't want to log that we're doing the horrible ruby stuff instead of blazing fast sql every time but it feels like we should somehow inform other models they might want to implement it.  Maybe?  I don't know.

Reply:
So I took a bit of time to think about this one.

The best idea I could come up with was to add some kind of rake task to check against every field that `MiqExpression` can be built against, and see if it can be supported in SQL for "includes all", "includes any", and "includes only".  That way, we can get a remaining list, and maybe farm these out at a "Hacktoberfest", or as we find time here and there.

Otherwise, I don't have a way to automate this.